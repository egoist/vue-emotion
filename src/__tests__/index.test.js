import { render, waitFor } from '@testing-library/vue'
import { VueEmotion, createGlobalStyle, styled } from '../index'

describe('VueEmotion', () => {
  it('createGlobalStyle', () => {
    const GlobalStyle = createGlobalStyle`
      body {
        background: red;
      }
    `
    const { container } = render({
      mixins: [VueEmotion],
      template: `
        <div>
          <GlobalStyle />
        </div>
      `,
      components: {
        GlobalStyle,
      }
    })

    const args = document.querySelectorAll('[data-emotion="css"]')
    expect(args[0].innerHTML).toMatchSnapshot()

    expect(container).toMatchSnapshot()
  })

  it('styled', () => {
    const Button = styled('button')`
      font-size: 15px;
    `

    const { container } = render({
      mixins: [VueEmotion],
      template: `
        <div>
          <Button class="emotion-vue">test</Button>
        </div>
      `,
      components: {
        Button,
      }
    })

    const args = document.querySelectorAll('[data-emotion="css"]')
    expect(args[1].innerHTML).toMatchSnapshot()

    expect(container).toMatchSnapshot()
  })
})
