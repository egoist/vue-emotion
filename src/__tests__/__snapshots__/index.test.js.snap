// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VueEmotion createGlobalStyle 1`] = `"body{background:red;}"`;

exports[`VueEmotion createGlobalStyle 2`] = `
<div>
  <div>
    <!---->
  </div>
</div>
`;

exports[`VueEmotion styled 1`] = `".css-fe74q7{font-size:15px;}"`;

exports[`VueEmotion styled 2`] = `
<div>
  <div>
    <button
      class="emotion-vue css-fe74q7"
    >
      test
    </button>
  </div>
</div>
`;
